src/components/Theme/src/Theme.vue

```diff
@@ -20,21 +21,66 @@ function isVueComponent(value) {
 	return value && value.render;
 }
 
+// returns true if value is an array
+// of objects with id fields
+function isKeyedArray(value) {
+	return value?.[0]?.id;
+}
+
+function keyedArrayIntoObject(arrayValue) {
+	const objectValue = {};
+	for (const item of arrayValue) {
+		objectValue[item.id] = item;
+	}
+	return objectValue;
+}
+
+function objectIntoKeyedArray(objectValue) {
+	const arrayValue = [];
+	for (const [id, item] of Object.entries(objectValue)) {
+		item.id = id;
+		arrayValue.push(item);
+	}
+	return arrayValue;
+}
+
+function mergeKeyedArrays(value, mergeValue) {
```

Comment:
We should consider adding some tests, the data manipulation here is sound but pretty complex. Having some tests to ensure we're always getting the right data with these methods would be ideal.

Reply:
can't write tests, we don't even have a unit test lib yet, and i think adding that would be outside the scope of this PR, so i created an issue for it: https://github.com/square/maker/issues/363